pkg/controller/core/workload_controller.go

```diff
@@ -345,13 +345,14 @@ func (r *WorkloadReconciler) reconcileMaxExecutionTime(ctx context.Context, wl *
 		return 0, nil
 	}
 
-	remainingTime := time.Duration(*wl.Spec.MaximumExecutionTimeSeconds)*time.Second - r.clock.Since(admittedCondition.LastTransitionTime.Time)
+	remainingTime := time.Duration(*wl.Spec.MaximumExecutionTimeSeconds-ptr.Deref(wl.Status.AccumulatedPastExexcutionTimeSeconds, 0))*time.Second - r.clock.Since(admittedCondition.LastTransitionTime.Time)
```

Comment:
I guess that this could negative value in the situations where the cluster or kueue-manager is higher load, or the kueue-manager restarted after the kueue-manager crashes.

Because in the situations, we can imagine `r.clock.Since(admittedCondition.LastTransitionTime.Time)` is a slightly larger value.

Reply:
In that case, can we do the below?

```suggestion
	remainingTime := max(time.Duration(*wl.Spec.MaximumExecutionTimeSeconds-ptr.Deref(wl.Status.AccumulatedPastExexcutionTimeSeconds, 0))*time.Second - r.clock.Since(admittedCondition.LastTransitionTime.Time), 0)
```

Reply:
I chatted with @trasc offline.
I assumed that the remainingTime will be stored in the .status.accumulatedPastExecutionTimeSeconds. So, I recommended going back to 0 in the case of a negative value.

But, in line 355, we reset the .status.accumulatedPastExecutionTimeSeconds when the value is negative.
Hence, my suggestion is not valuable.